main_ghost.py

```diff
@@ -69,40 +74,131 @@ def get_role(server_roles, target_id):
+        request = await botspam_channel.send(embed=request_embed)
+        await request.add_reaction("✅")
+        await request.add_reaction("❎")
+
+        def check(reaction, user):
+            return config.game_master in [y.id for y in user.roles] and reaction.message.id == request.id
+
+        try:
+            reaction_confirm, user = await client.wait_for('reaction_add', timeout=172800, check=check)
+        except asyncio.TimeoutError:
+            request_embed = discord.Embed(title="Quote Request [Timed Out]", description="Message from {} in <#{}> requested for quote by {}:".format(reaction.message.author.mention,reaction.message.channel.id,user.mention), color=0xff0000)
+            request_embed.add_field(name="Message Content", value="```" + reaction.message.content.replace('`', '`\u200B') + "```")
+            await request.edit(embed=request_embed)
+            await reaction_confirm.message.clear_reactions()
+        else:
+            if reaction_confirm.emoji == "✅":
+                request_embed = discord.Embed(title="Quote Request [Approved By {}]".format(user), description="Message from {} in <#{}> requested for quote by {}:".format(reaction.message.author.mention,reaction.message.channel.id,user.mention), color=0x00ff00)
+                request_embed.add_field(name="Message Content", value="```" + reaction.message.content.replace('`', '`\u200B') + "```")
+                await request.edit(embed=request_embed)
+                await reaction_confirm.message.clear_reactions()
+                quote_embed = discord.Embed(description=reaction.message.content, color=0x0000ff)
+                quote_embed.set_author(name=str(reaction.message.author), icon_url=reaction.message.author.avatar_url)
+                quote_embed.set_footer(text="{} | {} (UTC)".format(reaction.message.guild.name, reaction.message.created_at.strftime('%d %B %H:%M:%S')))
+                await quote_channel.send(embed=quote_embed)
+            if reaction_confirm.emoji == "❎":
+                stats.increment_stat("quotes_denied", 1)
+                request_embed = discord.Embed(title="Quote Request [Denied By {}]".format(user), description="Message from {} in <#{}> requested for quote by {}:".format(reaction.message.author.mention,reaction.message.channel.id,user.mention), color=0xff0000)
+                request_embed.add_field(name="Message Content", value="```" + reaction.message.content.replace('`', '`\u200B') + "```")
+                await request.edit(embed=request_embed)
+                await reaction_confirm.message.clear_reactions()
+
+
+# Whenever a message is edited
+@client.event
+async def on_message_edit(before, after):
+    if before.author == client.user: # We don't want to respond to our own edits
+        return
+    if before.content == after.content: # Ensure it wasn't just a pin
+        return
+
+    if before.id != after.id:
+        db.add_trash_message(after.id,after.channel.id)

```

Comment:
?

Reply:
What?

Reply:
What does it do?

Reply:
It records messages and saves them in the trash can table. This is in case a message is posted in a channel like the swamp or the wager channel, where it is necessary that the messages are cleaned every now and then.

Reply:
But this will never trigger???? Anyway, I guess the rest looks good to me

Reply:
I mean, if it doesn't, then okay. 😛 But I wanted to make sure.